actions/contacts/search.go

```diff
@@ -0,0 +1,64 @@
+package contacts
+
+import (
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet/actions"
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/mappings"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoin-sv/spv-wallet/server/auth"
+	"github.com/gin-gonic/gin"
+)
+
+// Search will fetch a list of contacts
+// Get contacts godoc
+// @Summary		Search contacts
+// @Description	Search contacts
+// @Tags		Contact
+// @Produce		json
+// @Param		page query int false "page"
+// @Param		page_size query int false "page_size"
+// @Param		order_by_field query string false "order_by_field"
+// @Param		sort_direction query string false "sort_direction"
+// @Param		conditions query string false "conditions"
+// @Success		200
+// @Router		/v1/contacts [get]
+// @Security	x-auth-xpub
+func (a *Action) search(c *gin.Context) {
```

Comment:
1. Will this `search` support querying for substrings for a autosuggestion feature?

![image](https://github.com/bitcoin-sv/spv-wallet/assets/152964795/3dbf50fb-9533-4f07-a368-16d441102487)

2. Does it support fetching all the contacts at once? - maybe if I don't provide page & page_size params?
3. Is the searching by params case sensitive? I'm wondering what should happen when a user have a contact "BoB@b.com" and he puts into paymail field "bob@b.com". I suppose the app should inform a user that it is a trusted contact.
![image](https://github.com/bitcoin-sv/spv-wallet/assets/152964795/7d2cdd77-ae43-4116-946c-cc7678c24c81)

Reply:
1. Good question. In my opinion, we should be consistent with other `/search` endpoints in the system for now.
2. Leave `page` and `page_size` as optional parameters for consistency, but DO NOT set any MAX_PAGE_SIZE on server side for now.
3. We sanitize paymails (at least we should, I'll ensure we do it during adding contact), so we should make sure the paymail condition is in lower case to not rely on database collation.

Reply:
> 1. Will this `search` support querying for substrings for a autosuggestion feature?
> 
> ![image](https://private-user-images.githubusercontent.com/152964795/310400367-3dbf50fb-9533-4f07-a368-16d441102487.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MzY3MTYsIm5iZiI6MTcwOTczNjQxNiwicGF0aCI6Ii8xNTI5NjQ3OTUvMzEwNDAwMzY3LTNkYmY1MGZiLTk1MzMtNGYwNy1hMzY4LTE2ZDQ0MTEwMjQ4Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNlQxNDQ2NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hOTg2Mzc0M2ViM2M0YmM3MjRmZTA4NWYzYjkxMjdmYzljNWE3MTZhZDJlYTliY2Q3ZTNlMTZkY2U3NTVkOTVmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.r6cBZlCfUPutVOpSCVeO8Sh-gHTieBFPmO2xDuhq_A0)
> 
> 2. Does it support fetching all the contacts at once? - maybe if I don't provide page & page_size params?
> 3. Is the searching by params case sensitive? I'm wondering what should happen when a user have a contact "[BoB@b.com](mailto:BoB@b.com)" and he puts into paymail field "[bob@b.com](mailto:bob@b.com)". I suppose the app should inform a user that it is a trusted contact. Or maybe the "paymail" should be a `special` field and it should always be transformed to lower case.
>    ![image](https://private-user-images.githubusercontent.com/152964795/310402089-7d2cdd77-ae43-4116-946c-cc7678c24c81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MzY3MTYsIm5iZiI6MTcwOTczNjQxNiwicGF0aCI6Ii8xNTI5NjQ3OTUvMzEwNDAyMDg5LTdkMmNkZDc3LWFlNDMtNDExNi05NDZjLWNjNzY3OGMyNGM4MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNlQxNDQ2NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xY2UwYzA0YTY4ZmQwNWM2NWVhMDEyMGQzYjFkNmJjNTE5YTRhMzhhNzJlZTM1OTc0ZjFjNmNiZTI0ZGFmZjk5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.AbxITLyMwbMl1A-Ej5zyt67lRDHDsC6sOrPM9_2SmNs)

1. It should work. It has the same logic under the hood as other `search`s. 
2. Yes. If you don't provide any metadata and params it should fetch all contacts.
3. Will add sanitizing paymails too